src/ACEtk/block/details/BaseTwoParameterDistribution.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef NJOY_ACETK_BLOCK_BASETWOPARAMETERDISTRIBUTION
+#define NJOY_ACETK_BLOCK_BASETWOPARAMETERDISTRIBUTION
+
+// system includes
+
+// other includes
+#include "ACEtk/EnergyDistributionType.hpp"
+#include "ACEtk/block/details/Base.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+namespace details {
+
+/**
+ *  @class
+ *  @brief Base class for a distribution using two parameters
+ */
+class BaseTwoParameterDistribution : protected details::Base {
+
+  /* fields */
+  double emin_;
+  double emax_;
+
+  /* auxiliary functions */
+  #include "ACEtk/block/details/BaseTwoParameterDistribution/src/verifySize.hpp"
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/details/BaseTwoParameterDistribution/src/ctor.hpp"
+
+  /**
+   *  @brief Return the distribution type
+   */
+  static constexpr EnergyDistributionType LAW() {
+
+    return EnergyDistributionType::LevelScattering;
+  }
+
+  /**
+   *  @brief Return the distribution type
+   */
+  static constexpr EnergyDistributionType type() {
+
+    return EnergyDistributionType::LevelScattering;
+  }
+
+  /**
+   *  @brief Return the minimum incident energy for the distribution
+   */
+  double minimumIncidentEnergy() const { return this->emin_; }
+
+  /**
+   *  @brief Return the maximum incident energy for the distribution
+   */
+  double maximumIncidentEnergy() const { return this->emax_; }
+
+  /**
+   *  @brief Return the value of C1 = ( A + 1 ) / A * abs( Q )
+   */
+  double C1() const { return this->XSS( 1 ); }
+
+  /**
+   *  @brief Return the value of C2 = ( A / ( A + 1 ) )^2
```

Comment:
These comments seem to be for a specific form of this distribution, but for example DiscretePhotonDistribution, C2 is the photon binding energy. Same applies to above.

Reply:
I made this into a base class, the comments must have survived the refactoring